docs/tutorials/dataassociation/KDTree_Tutorial.py

```diff
@@ -0,0 +1,508 @@
+#
+# Fig. 1a offers a graphical representation of the :math:`k`-d tree shown in 1b. :math:`d` represents the depth of
+# the tree. The :math:`k`-d tree is constructed from a series of detections, which in this case are 2-dimensional
+# vectors, with splits alternating in each dimension. At :math:`d=0` the split occurs in the :math:`x`-axis, at
+# :math:`d=1` the split occurs in the :math:`y`-axis etc. Colours of the splitting planes in 1a correspond with the
+# splits at each depth shown in 1b.
+#
+# In comparison, entries in TPR tree leaf nodes are pairs of a moving object's position and an ID for the object. The
+# internal nodes of the tree are bounding rectangles which bound the positions of all moving objects (or other bounding
+# rectangles) in that subtree. The bounding rectangles' coordinates are functions of time, so they are capable of
+# following the enclosed data points as they move over time.
+
+
+# %%
+# Searching a :math:`k`-d tree with Nearest Neighbour
+# ---------------------------------------------------
+# Once the :math:`k`-d tree has been constructed from the detection set, it can then be used for data association. In
+# this example we will search for the detection that is the nearest neighbour of a given prediction.
+#
+# The method for searching the tree is similar to the method for building the tree: starting at the root node, the
+# :math:`x^{th}` dimension of the prediction is compared with the :math:`x^{th}` dimension of the root node detection.
+# If the value of the prediction's :math:`x` dimension is less than or equal to that of the root node detection, the
+# search moves down the left subtree, and down the right subtree if the value is greater. This process is repeated
+# recursively at each internal node, again alternating through the vector dimensions, until a leaf node is reached.
+# Because we are conducting a Nearest Neighbour search, at each node the Euclidean distance between the prediction and
+# the detection at that node is measured. If the distance is less than the current best distance, the detection at that
+# node becomes the new current best nearest neighbour.
+#
+# To ensure that the true nearest neighbour has been identified, once we reach a leaf node the search must continue back
+# up through the tree to ensure we haven't skipped a potential nearest neighbour that lies in an over-looked sub-space.
+# As the search moves back up the tree, we should at each level consider if we should go down the sibling subtree.
+#
+# .. image:: ../../_static/kd_tree_fig_2.png
+#   :width: 1200
+#   :height: 250
+#   :alt: Image showing diagrammatical representation of kd-tree in 2 dimensions, described further in figure 2 caption.
+#
+# Fig. 2a offers a graphical representation of the :math:`k`-d tree shown in 2b. 2a shows the prediction vector,
+# :math:`[8,2]`, in blue. 2b shows the first part of the search taking place down the :math:`k`-d tree. Current best
```

Comment:
In the constructing a k-d tree section you state a vector space of [x, y]^T using :math:`\begin{bmatrix} x \\ y \\\end{bmatrix}`. However for the rest of the document each vector is displayed as math:`[x, y]`. It might be best for continuity/mathematical correctness to display each vector as math:`[x, y]^T` to denote that each is a vector in the same form as previously stated. You could use :math:`\begin{bmatrix} x \\ y \\\end{bmatrix}` throughout but it might make the lines break up a bit.
```suggestion
# :math:`[8,2]^T`, in blue. 2b shows the first part of the search taking place down the :math:`k`-d tree. Current best
```